components/ws-manager-api/core.proto

```diff
@@ -598,3 +613,9 @@ message ExposedPorts {
     // ports is the set of ports which ought to be exposed to the internet
     repeated PortSpec ports = 1;
 }
+
+// ExposedPorts describes the user uploaded ssh public keys, it will be use only in annotations.
+message SSHPublicKeys {
+    // keys is the set of ssh public key
+    repeated string keys = 1;
```

Comment:
This is redundant indeed. Also, I'm wondering why there's no `keys` attribute on teh `StartWorkspaceSpec`